src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,380 @@
+      val correctNewPeer = newProps.sortBy { case (id, snap)  => (-snap.height, snap.hash) }.map(_._1).head
+      res.sortedSnaps.minBy { case snap => -snap.height } shouldBe correctNewSnaps
+      res.nodeIdsWithSnaps shouldBe Set(correctNewPeer)
+    }
+
+    "should return empty diff if not enough snaps for a majority" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.foreach {
+        case (id, peerApi) =>
+          peerApi.client.getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF Array()
+      }
+      val updateProps = redownloadService.fetchAndSetPeerProposals()
+      val newMajority = redownloadService.recalculateMajoritySnapshot()
+      val res = (updateProps >> newMajority).unsafeRunSync()
+      res.sortedSnaps shouldBe Seq()
+      res.nodeIdsWithSnaps shouldBe Set()
+    }
+  }
+
+  "checkForAlignmentWithMajoritySnapshot" - {
+    "should trigger download if should redownload" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      val newSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.foreach {
+        case (_, peerApi) =>
+          peerApi.client
+            .getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF serializedResponse
+      }
+      val updateProps = redownloadService.fetchAndSetPeerProposals()
+      val persist = redownloadService.persistLocalSnapshot(newSnapshot)
+      val check = redownloadService.checkForAlignmentWithMajoritySnapshot()
+      val res = (updateProps >> persist >> check).unsafeRunSync()
+      res shouldBe List(6, 4, 2, 0).map(toRecentSnapshot).some
+    }
+
+    "should not trigger download if aligned with majority" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      val newSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
+      val persist = redownloadService.persistLocalSnapshot(newSnapshot)
+      val check = redownloadService.checkForAlignmentWithMajoritySnapshot()
+      val res = (persist >> check).unsafeRunSync()
+      res shouldBe None
+    }
+  }
+
+  "shouldReDownload" - {
+    val height = 2
+    val ownSnapshots = List(height).map(i => RecentSnapshot(s"$i", i, Map.empty))
+    val interval = RedownloadService.snapshotHeightRedownloadDelayInterval
+
+    "should return true when there are snaps to delete and to download" - {
+      val diff =
+        SnapshotDiff(
+          List(RecentSnapshot("someSnap", height, Map.empty)),
+          List(RecentSnapshot("someSnap", height, Map.empty)),
+          List(Id("peer"))
+        )
+
+      RedownloadService.shouldReDownload(ownSnapshots, diff, facilitators.keySet) shouldBe true
```

Comment:
Why some methods are moved to companion object? I guess it should be just instantiated and it's dependencies should be mocked 🤔 

Just asking because I rarely use companion objects (only when some constants are shared between instances etc). Maybe there is some pattern for that :P